scripts/ci/docker-test.sh

```diff
@@ -87,6 +87,8 @@ print_logs () {
 }
 
 declare -i max_restore_container_tries=3
+# $current_iteration is used to track the number of times we have tried to
+# restore a container. It is reset to 0 when a restore succeeds.
 current_iteration=
```

Comment:
pls replace a recursive call with a loop and move current_iteration in the restore_container function.